src/components/RunGraph.vue

```diff
@@ -1,13 +1,35 @@
 <template>
-  <div class="run-graph">
-    hello world
-  </div>
+  <canvas ref="canvas" />
 </template>
 
 <script lang="ts" setup>
+  import { onMounted, ref } from 'vue'
   import { RunGraphConfig } from '@/models/RunGraph'
+  // eslint-disable-next-line import/default
+  import PixiWorker from '@/workers/worker.ts?worker'
 
   defineProps<{
     config: RunGraphConfig,
   }>()
+
+  const canvas = ref<HTMLCanvasElement>()
+
+  onMounted(() => {
+    if (!canvas.value) {
+      throw new Error('Canvas does not exist')
+    }
+
+    const view = canvas.value.transferControlToOffscreen()
+    const worker = new PixiWorker()
+
+    // this isn't type safe
+    worker.postMessage({
```

Comment:
If you'll be communicating frequently with the worker, it might be helpful to create an intermediary function that _is_ typesafe like we do [here](https://github.com/PrefectHQ/nebula-ui/blob/main/src/workers/util/promise-channel.ts) - certainly some room for error but it could save you debug time

Reply:
Yeah I'll implement something like that probably. Though I'm not sure yet if or how much communication will go back and forth. I'm sure there will be some pixi events that will need to be emitted through the component.